client/state/reader/teams/actions.js

```diff
@@ -7,11 +7,26 @@
  */
 import {
 	READER_TEAMS_REQUEST,
+	READER_TEAMS_RECEIVE,
 } from 'state/action-types';
 
 export function requestTeams() {
 	return {
 		type: READER_TEAMS_REQUEST,
+		meta: {
+			requestStart: READER_TEAMS_REQUEST
+		}
+	};
+}
```

Comment:
> helper func possibility: serialize the whole action

One thing that comes to mind is that both the keys for request and receive actions need to match. Most receive actions just have the type and item, so adding any request params onto the signature would be a change.

My intuition is that we should keep the actions rather simple and straightforward, while we wrap the meta at a higher layer. The data layer can maybe pass through meta props we're still interested in.

A quick thought:
```javascript
// maybe in the query component?
requestStart( key, requestAction() );
//in the data-layer?
const dataHandler = ( { dispatch }, action ) => {
    const key = get( action, 'meta.requestStart' );
    //...
   return promise.then( () => {
        dispatch( mySuccessAction() );
        dispatch( requestEnd( key, myReceiveAction() ) );
    } )
    .catch( () => {
        dispatch( requestEnd( key, myErrorAction() ) );
    } );
}
```